documentation-site/components/yard/const.ts

```diff
@@ -0,0 +1,101 @@
+import {KIND, SIZE, SHAPE} from 'baseui/button';
+
+export enum Action {
+  UpdateCode,
+  UpdatePropsAndCode,
+  Reset,
+}
+
+export enum PropTypes {
+  String = 'string',
+  ReactNode = 'react node',
+  Boolean = 'boolean',
+  Number = 'number',
+  Enum = 'enum',
+  Array = 'array',
+  Object = 'object',
+  Function = 'function',
+  Overrides = 'overrides',
+}
+
+export const COMPONENTS = {
+  Button: {
```

Comment:
should we move this part to a new file for the button component? so for the next component, we'd show the implementer the structure we'd like to have, instead of making this file really long?

Reply:
Yep, this is a temporary location, still in the process of figuring out of the ideal structure. Once this works for multiple components + theming values and we are ready to roll it out for all components, we can colocate it with the components. 

I was even thinking that it could be exported so userland could build their own tooling / documentation based on this data-structure.